src/api/bulkQuotes/{id}.js

```diff
@@ -0,0 +1,113 @@
+const Config = require('../../lib/config')
+const dto = require('../../lib/dto')
+
+const { kafkaConfig } = new Config()
+
+/**
+ * Operations on /bulkQuotes/{id}
+ */
+module.exports = {
+  /**
+     * summary: getBulkQuotesById
+     * description: The HTTP request GET /bulkQuotes/&lt;id&gt; is used to get information regarding an earlier created or requested bulk quote. The &lt;id&gt; in the URI should contain the bulkQuoteId that was used for the creation of the bulk quote.
+     * parameters: Accept
+     * produces: application/json
+     * responses: 202, 400, 401, 403, 404, 405, 406, 501, 503
+     */
+  get: async function getBulkQuotesById (context, request, h) {
+    const histTimerEnd = Metrics.getHistogram(
+      'bulkQuotes_id_get',
+      'Publish HTTP GET /bulkQuotes/{id} request',
+      ['success']
+    ).startTimer()
+
+    try {
+      logger.debug('got a GET /bulkQuotes request: ', request.payload)
+      const { topic, config } = kafkaConfig.PRODUCER.BULK_QUOTE.GET
+      const topicConfig = dto.topicConfigDto({ topicName: topic })
+      const message = dto.messageFromRequestDto(request, Events.Event.Type.BULK_QUOTE, Events.Event.Action.GET)
+
+      await Producer.produceMessage(message, topicConfig, config)
+
+      histTimerEnd({ success: true })
+    } catch (err) {
+      logger.error(`error in GET /bulkQuotes request: ${err?.message}`)
+      // todo: think, how we should handle such error cases:
+      //   - how to send callback ?
+      //   - OR reply with errorCode (not 202)?
+      histTimerEnd({ success: false })
+    }
+
+    return h.response().code(Http.ReturnCodes.ACCEPTED.CODE)
+  },
+  /**
+     * summary: putBulkQuotesById
+     * description: The callback PUT /bulkQuotes/&lt;id&gt; is used to inform the client of a requested or created bulk quote. The &lt;id&gt; in the URI should contain the bulkQuoteId that was used for the creation of the bulk quote, or the &lt;id&gt; that was used in the GET /bulkQuotes/&lt;id&gt;.
+     * parameters: body, Content-Length
+     * produces: application/json
+     * responses: 200, 400, 401, 403, 404, 405, 406, 501, 503
+     */
+  put: async function putBulkQuotesById (context, request, h) {
+    const histTimerEnd = Metrics.getHistogram(
+      'bulkQuotes_id_put',
+      'Publish HTTP PUT /bulkQuotes/{id} request',
+      ['success']
+    ).startTimer()
+
+    try {
+      logger.debug('got a PUT /bulkQuotes request: ', request.payload)
+      const { topic, config } = kafkaConfig.PRODUCER.BULK_QUOTE.PUT
+      const topicConfig = dto.topicConfigDto({ topicName: topic })
+      const message = dto.messageFromRequestDto(request, Events.Event.Type.BULK_QUOTE, Events.Event.Action.PUT)
+
+      await Producer.produceMessage(message, topicConfig, config)
+
+      histTimerEnd({ success: true })
+    } catch (err) {
+      logger.error(`error in PUT /bulkQuotes request: ${err?.message}`)
+      // todo: think, how we should handle such error cases:
```

Comment:
Believe we would just rethrow the error and have a http sycronous 500 be sent back.
The kafka handler should handle the sending of the error callback notification if something goes wrong processing wise.

Reply:
...rethrow the error and have a http syncronous 500 be sent back...
@kleyow, Initially I also was thinking about it, but when I checked Swagger API docs, I didn't find 500 status code among allowed response for quotes/bulkQuotes (only 501 and 503, which is not really suitable for publishing error, IMHO).

Reply:
Hmm...I'm just only now realizing that. I'm not sure how to answer now.